### ISSUE TITLE
Add Pdf.js support for progressive loading of pdf.

### ISSUE THREAD

**Issue by loftux** [share-extras/media-viewers#1]
pdf.js has implemented support for progressive loading of pdf.
https://github.com/mozilla/pdf.js/pull/2719

I will create a branch to try this feature out.

TBD if to add progress bar when loading large pdf:s

**Comment by wabson** [comment]
Peter, did you create a branch for this feature in the end?

**Comment by loftux** [comment]
Yes, I've created a pdfjs branch where i pull in all the latest from pdf.js project.
So you can check out that branch to test. I just updated it to the latest from Mozilla pdf.js.

The progressive loading is disabled by default. To test out those features you manually need to update in PdfJs.js
Current settings
         // PDFJS Disable new range feature if not loading from first page
         PDFJS.disableRange = true;
         // disable autofetch - retrieve just the ranges needed to display
         PDFJS.disableAutoFetch = false;
I added them as there were issues if pdf was not loaded from page 1 (ie if you had viewed the page previously and the pdf tried to load from stored page number). This may have been fixed now, haven't tested that.

Set disableRange to false to test.
disableAutoFetch = true will as I understand just fetch the range that is needed for display.

In Safari there is a bug, so PDFJS.disableRange is set to true in compatibility.js, if we move forward to implement this we must make sure we do not overwrite this for Safari. https://github.com/mozilla/pdf.js/issues/3352

**Comment by loftux** [comment]
I've now added configuration support.
However, I'm not sure we should keep the Module configuration option in final release. There are already a lot of viewer configuration options and this will only confuse.
Add to this that in most cases the loading of the pdf is slower. 

Besides enabling the loading, we have to rewrite the pdf page rendering for this to become useful, and that is part of next milestone.

**Comment by wabson** [comment]
Agreed. The module is useful for developers experimenting with progressive loading but it is yet another one that will show up in the console.

What about giving the extension module file a `.sample` suffix, then we leave it in there but it would be disabled for now?

**Comment by loftux** [comment]
I removed the config for now, and changing milestone to next interation.
